### DISCUSSION TITLE
Missing example(s) for Alpine.

### DISCUSSION THREAD

**Discussion by nerdoc** [tetra-framework/tetra#30] [Ideas]
It's very unclear how Alpine uses the data, especially directives like x-show, which is extremely useful. You should definitively add a section to the docs that helps with Alpine examples.

I made a short component, just to test functionality:

```python
@default.register
class MyComponent(Component):
    visible = public(True)

    @public
    def show(self):
        self.visible = True

    @public
    def hide(self):
        self.visible = False

    template: django_html = """
    <div>
        <button class="btn btn-secondary" x-on:click="visible = ! visible">Toggle with Alpine</button>
        <button class="btn btn-secondary" @click="show()">Show using server code</button>
        <button class="btn btn-secondary" @click="hide()">Hide using server code</button>
        <div x-show="visible">
            ...content...
        </div>
    </div>
    """
```
I had to find out using try&error...
Maybe you find a better example here, but I think this will help newcomers to find into the code.

And one more thing, could you add a search form on the docs page? That's urgently needed.

**Comment by samwillis** [comment]
Thanks!

Yes, the plan is to move the docs to Sphinx/Furo and add search.

**Reply by nerdoc** [reply]
help needed?

**Reply by samwillis** [reply]
If you have the time and would like to help, that would be brilliant. I investigated both Sphinx and MkDocs a few months back, and I think I prefer Sphinx (I forget exactly why now), and the [Furo](https://pradyunsg.me/furo/) theme looks like a good starting point and had lots of nice features like [admonitions](https://pradyunsg.me/furo/reference/admonitions/).

A lot of project seem to be going with MkDocs and the Material theme https://squidfunk.github.io/mkdocs-material/ but I find it a very busy and distracting theme for a docs site, I much prefer the simplicity of Furo.

**Reply by nerdoc** [reply]
I am using sphinx for a lot of projects, and am super satisfied with it. I am using readthedocs.io - works very well. I must admit I was using .rst instead of .md, but I dislike the syntax very much, Markdown is much easier, especially with links.
Only thing is the index and cross-links - but maybe md is sufficient here too.
If you want I can setup a page at readthedocs.io - I can transfer the account to you when it's "finished". I'd first do that in a fork - I think it's not a big deal later to switch the autogenerated docs to your repo later.

**Reply by nerdoc** [reply]
I tested both, and meanwhile think that MkDocs the easier and straightforward way of doing the tetra documentation, it "speaks" markdown natively, and the Material plugin can be made simple looking as well.
![grafik](https://github.com/tetra-framework/tetra/assets/2955584/7a1a3192-51a7-46a6-98e6-b4fc6495810c)

So, i have it up and running locally, and would add it to readthedocs shortly. If @TheCulliganMan gives me the handle for RTD's "tetra", I'll put it there, else I'll use "tetraframework".
